content/en/blog/_posts/2021-04-08-kubernetes-release-1.21.md

```diff
@@ -0,0 +1,120 @@
+
+### Immutable Secrets and ConfigMaps
+Immutable Secrets and ConfigMaps adds a new field to those resource types which will reject changes to those objects if set. Secrets and ConfigMaps by default are mutable which is beneficial for pods that are able to consume changes. Mutating Secrets and ConfigMaps can also cause problems if a bad configuration is pushed for pods that use them.
+
+By marking Secrets and ConfigMaps as immutable you can be sure your application configuration won't change. If you want to make changes you'll need to create a new, uniquly named Secret or ConfigMap and deploy a new pod to consume that resource. Immutable resources also have scaling benefits because controllers do not need to poll the API server to watch for changes.
+This feature is now graduating to stable in Kubernetes 1.21.
+
+### Graceful Node Shutdown
+Graceful Node shutdown is graduating to beta with this release (and will now be available to a much larger group of users!). This is a hugely beneficial feature that allows Kubelet to be aware of Node shutdown, and gracefully terminate Pods that are scheduled to that Node.
+
+Currently, when a node shuts down, Pods do not follow the expected termination lifecycle and are not shutdown gracefully. This can introduce problems with a lot of different workloads. Going forward, Kubelet will be able to propagate the shutdown signal to Pods ensuring they can terminate as gracefully as possible.
+
+### PV Health Monitor
+Physical Volumes (PV) are commonly used in applications to get local, file-based storage. They can be used in many different ways and help users migrate applications without needing to re-write storage backends.
+
+Kubernetes 1.21 has a new alpha feature which allows PVs to be monitored for health of the volume and marked accordingly if the volume becomes unhealthy. Workloads will be able to react to the health state to protect data from being written or read from a volume that is unhealthy.
+
+### Reducing Kubernetes Build Maintenance
+Previously Kubernetes has maintained multiple build systems. This has often been a source of friction and complexity for new and current contributors.
+
+Over the last release cycle, a lot of work has been put in to simplify the build process, and standardize on the native Golang build tools. This should empower broader commmunity maintenance, and lower the barier to entry for new contributors.
+
+## Major Changes
+
+### PodSecurityPolicy Deprecation
+As of Kubernetes 1.21 PodSecurityPolicies in their current form will be deprecated. Even though the feature has been deprecated, it will continue to be available and function as before until Kubernetes 1.25. Going forward, there is a proposal in place for a simplified feature that will fill the basic use cases of PodSecurityPolicies.
+
+SIG security has prepared a dedicated blog post with more details about the PodSecurityPolicy feature deprecation. You can read that [here.]()
+
```

Comment:
@divya-mohan0209 - @IanColdwater has suggested changing this section to:

> In Kubernetes 1.21, PodSecurityPolicy is deprecated. As with all Kubernetes feature deprecations, PodSecurityPolicy will continue to be available and fully-functional for several more releases. PodSecurityPolicy is currently planned for remvoal in Kubernetes 1.25.

> What's next? We're developing a new feature, tentatively called PodIsolationPolicy, that covers the key PodSecurityPolicy use cases with greatly improved ergonomics and maintainability. To learn more, check out the PSP feature blog [here](link to the larger blog).

Reply:
Presenting the same text as Markdown:
```markdown
In Kubernetes 1.21, PodSecurityPolicy is deprecated. As with all Kubernetes
feature deprecations, PodSecurityPolicy will continue to be available and fully-functional for
several more releases. PodSecurityPolicy is planned for removal in Kubernetes 1.25.

What's next? We're developing a new feature, tentatively called PodIsolationPolicy,
that covers the key PodSecurityPolicy use cases with greatly improved ergonomics and
maintainability. To learn more, check out the PSP feature blog [here](link to the larger blog).
```
I also fixed a typo and tweaked a word (“currently”).

I recommend using the _title_ of the other article as the link text.

Reply:
Once we have the blog name, I shall have the [here] reworded.